lib/semantics/check-declarations.cpp

```diff
@@ -295,6 +295,10 @@ void CheckHelper::CheckValue(
 
 void CheckHelper::CheckObjectEntity(
     const Symbol &symbol, const ObjectEntityDetails &details) {
+  if (symbol.test(Symbol::Flag::LocalityLocal) ||
```

Comment:
What happens when new kinds of compiler-created symbols are invented?

For future-proofing, consider defining a member function or predicate on `Symbol` that determines whether the symbol is compiler-created, copying this Boolean expression into that function, and using it here.

Reply:
Yes.  I had that same thought after I pushed this change.  I'll do it.  But first I'll see if there are other instances where the compiler creates entities.

Reply:
It seems to me these should be HostAssocDetails symbols rather than new object symbols. Then we wouldn't be seeing them here at all.

Reply:
Thanks, Tim.  I'll look into doing that.

Reply:
I've pushed a new commit where I created host associated symbols for names that appear in LOCAL and LOCAL_INIT locality-specs.